File: src/model_v4/entities/SBS_APKLASSE.ts
import { Column, Entity, Index, JoinColumn, ManyToOne, OneToMany, PrimaryColumn } from "typeorm";
import { SBS_AP } from "./SBS_AP";
import { SBS_APTYP } from "./SBS_APTYP";
import { SBS_EXTPROG } from "./SBS_EXTPROG";


@Entity("SBS_APKLASSE",{schema:"sbsdb"})
@Index("FK872BB9CF9E420B01",["aPTYP_INDEX",])
export class SBS_APKLASSE {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"APKLASSE_INDEX"
        })
    APKLASSE_INDEX:string;
        

    @Column("varchar",{ 
        nullable:false,
        length:50,
        name:"APKLASSE"
        })
    APKLASSE:string;
        

    @Column("bigint",{ 
        nullable:true,
        default: () => "'NULL'",
        name:"FLAG"
        })
    FLAG:string | null;
        

   
    @ManyToOne(type=>SBS_APTYP, SBS_APTYP=>SBS_APTYP.sBS_APKLASSEs,{ onDelete: 'RESTRICT',onUpdate: 'RESTRICT' })
    @JoinColumn({ name:'APTYP_INDEX'})
    aPTYP_INDEX:SBS_APTYP | null;


   
    @OneToMany(type=>SBS_EXTPROG, SBS_EXTPROG=>SBS_EXTPROG.aPKLASSE_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_EXTPROGs:SBS_EXTPROG[];
    

   
    @OneToMany(type=>SBS_AP, SBS_AP=>SBS_AP.aPKLASSE_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_APs:SBS_AP[];
    
}

File: SbsdbServer/sbsdbv4/model/SbsApklasse.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model
{
    [Table("SBS_APKLASSE")]
    public partial class SbsApklasse
    {
        public SbsApklasse()
        {
            SbsAp = new HashSet<SbsAp>();
            SbsExtprog = new HashSet<SbsExtprog>();
        }

        [Column("APKLASSE_INDEX", TypeName = "bigint(20)")]
        public long ApklasseIndex { get; set; }
        [Required]
        [Column("APKLASSE", TypeName = "varchar(50)")]
        public string Apklasse { get; set; }
        [Column("FLAG", TypeName = "bigint(20)")]
        public long? Flag { get; set; }
        [Column("APTYP_INDEX", TypeName = "bigint(20)")]
        public long? AptypIndex { get; set; }

        [ForeignKey("AptypIndex")]
        [InverseProperty("SbsApklasse")]
        public virtual SbsAptyp AptypIndexNavigation { get; set; }
        [InverseProperty("ApklasseIndexNavigation")]
        public virtual ICollection<SbsAp> SbsAp { get; set; }
        [InverseProperty("ApklasseIndexNavigation")]
        public virtual ICollection<SbsExtprog> SbsExtprog { get; set; }
    }
}
